activation/handler_v2_test.go

```diff
@@ -647,20 +645,27 @@ func marryIDs(
 
 func TestHandlerV2_ProcessMergedATX(t *testing.T) {
```

Comment:
Maybe add a test for multiple merged ATXs created from different marriage sets to ensure the function can distinguish between valid and invalid merges.

Reply:
I don't understand what you have in mind, could you give an example?

Reply:
The idea is to write a test which ensures that the function can distinguish between valid and invalid merges, correctly identifying and handling cases where multiple ATXs share the same MarriageATX.
Example: 
- Create Marriage ATX IDs (marriageATX1 and marriageATX2)
marriageATX1 := ATXID{A, B, C}.       marriageATX2 := ATXID{D, E, F}
- Creates two ActivationTx instances (atx1 and atx2) with different MarriageATX IDs.
- Add ATXs to database and generate Checkpoint. 
- Creates a new ActivationTx (invalidATX) with the same MarriageATX as atx1.
- Adds invalidATX to the database and try to generate the checkpoint again (error message "invalid merge")

Reply:
>Creates a new ActivationTx (invalidATX) with the same MarriageATX as atx1.

What would be the reason that this ATX is invalid?

>Adds invalidATX to the database and try to generate the checkpoint again (error message "invalid merge")

It sounds as if the checkpoint generation was supposed to fail. Is this the intention?

Reply:
We persist malicious ATXs in the DB so the checkpoint generation could come across 2 ATXs with the same epoch and marriage ATX. I think it shouldn't fail then.

Reply:
Oh I see, so it should correctly identify such a case by marking or flagging them as malicious rather than invalid.

Reply:
Yes, this is covered by checkpoint generation tests. The ATXs of malicious smeshers are not included in the generated checkpoint.